crates/storage-macros/tests/compile_fail/atomic_unit.stderr

```diff
@@ -0,0 +1,10 @@
+error[E0308]: mismatched types
+  --> tests/compile_fail/atomic_unit.rs:28:30
+   |
+28 |         let _: Vec<Parent> = interface.children_of(&parent).unwrap();
+   |                -----------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Vec<Parent>`, found `Vec<Child>`
+   |                |
+   |                expected due to this
+   |
+   = note: expected struct `Vec<Parent>`
+              found struct `Vec<Child>`
```

Comment:
🤔

Reply:
yeah, should this be matched for? this isn't a macro-emitted error

Reply:
Correct, it's not macro-emitted. This is a compilation error test, i.e. ensuring that something that is expecting to fails, fails. It's there to check that the wrong type is not accepted, to ensure that the constraints are correct.

Reply:
Isn't that already accounted for with the first case?

```rust
        let _: Vec<Child> = interface.children_of(&parent).unwrap();
```

if this compiles, it's guaranteed to be a `Vec<Child>` is it not?

Reply:
No, because `Child` implements certain traits and this is a constraint check.